src/lib.rs

```diff
@@ -1655,4 +1701,43 @@ mod tests {
         v.push(true);
         assert_eq!(v.vec.len(), 1);
     }
+
+    #[test]
+    fn to_bytes_1_byte_a() {
+        let x = Vob::from_elem(8, true);
+        assert_eq!(x.to_bytes(), [255]);
+    }
+    #[test]
+    fn to_bytes_1_byte_b() {
+        let mut x = Vob::from_elem(8, true);
+        x.set(1, false);
+        assert_eq!(x.to_bytes(), [0b10111111]);
+    }
+
+    #[test]
+    fn to_bytes_empty() {
+        let x = Vob::new();
+        assert_eq!(x.to_bytes(), []);
+    }
+
+    #[test]
+    fn to_bytes_2_bytes() {
+        let mut x = Vob::from_elem(9, false);
+        x.set(2, true);
+        x.set(8, true);
+        assert_eq!(x.to_bytes(), [32, 128]);
+    }
+
+    #[test]
+    fn to_bytes_using_binary_a() {
+        let x = Vob::from_elem(1, true);
+        assert_eq!(x.to_bytes(), [0b10000000]);
+    }
+
+    #[test]
+    fn to_bytes_using_binary_b() {
```

Comment:
Ditto.

Reply:
Hi, I've made the majority of the changes.
I didn't try to change the implementation of to_bytes to use a bit array as you suggested.
For the tests, I removed the ones you mentioned and added a test to test to_bytes on a Vob with more than one word, it's a simple test because I wasn't sure on how to create a test that tests to_bytes on random Vobs. I made all the other changes you have mentioned though